### PR TITLE
Add `fail-fast` option to test runner

### PR DESCRIPTION
**Pull request by dwoz**
### What does this PR do?

Adds `--fail-fast (-f)` option to test suite. This will cause the test runner to exit on the first failure. 

### Commits signed with GPG?

Yes

### CODE REVIEW

File: tests/support/parser/__init__.py [hunk 1]
```diff
@@ -241,6 +241,14 @@ def __init__(self, testsuite_directory, *args, **kwargs):
         self.output_options_group = optparse.OptionGroup(
             self, 'Output Options'
         )
+        self.output_options_group.add_option(
+            '-f',
```
Comment (cachedout) [inline tests/support/parser/__init__.py]: I wonder if this should be `-F`? I always think of the lowercase as `force`.

Comment (dwoz) [reply tests/support/parser/__init__.py]: I'm cool with that and will push this minor change.

**Comment by rallytime** [comment]
re-run py

**Comment by rallytime** [comment]
@dwoz Something in here isn't quite right. There are only ~20 tests running with this change. It must be conflicting somewhere. Can you take a look?

**Comment by dwoz** [comment]
@rallytime Yes, I'll check it out. Thanks for the help here.

File: tests/support/parser/__init__.py (hunk 1 above)
Comment (dwoz) [reply tests/support/parser/__init__.py]: @cachedout Pytest uses `-X` not sure if that changes anything on your thoughts here or not. :)

https://docs.pytest.org/en/latest/usage.html#stopping-after-the-first-or-n-failures

**Review by rallytime — APPROVED** [review]

**Pull request merged**